beacon-chain/rpc/beacon/assignments.go

```diff
@@ -49,18 +40,24 @@ func (bs *Server) ListValidatorAssignments(
 		requestedEpoch = q.Epoch
 	}
 
-	if requestedEpoch > helpers.CurrentEpoch(headState) {
+	currentEpoch := helpers.SlotToEpoch(bs.GenesisTimeFetcher.CurrentSlot())
+	if requestedEpoch > currentEpoch {
 		return nil, status.Errorf(
 			codes.InvalidArgument,
 			"Cannot retrieve information about an epoch in the future, current epoch %d, requesting %d",
-			helpers.CurrentEpoch(headState),
+			currentEpoch,
 			requestedEpoch,
 		)
 	}
 
+	requestedState, err := bs.StateGen.StateBySlot(ctx, helpers.StartSlot(requestedEpoch))
```

Comment:
why not just use this over here ?
```
bs.GenesisTimeFetcher.CurrentSlot()
```

Reply:
User has option to choose to request an arbitrary slot that's less than current slot. See L39

```go
case *ethpb.ListValidatorAssignmentsRequest_Epoch:
		requestedEpoch = q.Epoch
	}
```